File: practice/mathematics/algebra/little_guarav_and_sequence/little_guarav_and_sequence.py
#!/usr/bin/env python

def log2(i_n):
    res = 0
    while i_n > 0:
        res += 1
        i_n //= 2
    return res - 1

t = int(raw_input().strip())
for _ in xrange(t):
    n = int(raw_input().strip())
    l = log2(n)
    
    rem1 = 1 if n % 2 == 0 else 5
    rem2 = 2 if l == 0 else 6 * l % 10
    print rem1 * rem2 % 10

File: practice/mathematics/algebra/little_guarav_and_sequence/little_guarav_and_sequence.cpp
#include <iostream>

using namespace std;

unsigned log_2(unsigned long long i_val)
{
    unsigned res = 0;
    while (i_val > 0)
    {
        res++;
        i_val >>= 1;
    }
    return res - 1;
}

int main()
{
    unsigned t;
    unsigned long long n;
    
    cin >> t;
    while (t--)
    {
        cin >> n;
        unsigned lg = log_2(n);
        
        unsigned rem1 = n % 2 == 0 ? 1 : 5;
        unsigned rem2 = lg == 0 ? 2 : (6 * lg) % 10;
        unsigned res = (rem1 * rem2) % 10;
        cout << res << endl;
    }
    
    return 0;
}
